### PR TITLE
[SPARK-27760][CORE] Spark resources - change user resource config from .count to .amount

### PR DESCRIPTION
**Pull request by tgravescs**
## What changes were proposed in this pull request?

Change the resource config spark.{executor/driver}.resource.{resourceName}.count to .amount to allow future usage of containing both a count and a unit.  Right now we only support counts - # of gpus for instance, but in the future we may want to support units for things like memory - 25G. I think making the user only have to specify a single config .amount is better then making them specify 2 separate configs of a .count and then a .unit.  Change it now since its a user facing config.

Amount also matches how the spark on yarn configs are setup.

## How was this patch tested?

Unit tests and manually verified on yarn and local cluster mode

### CODE REVIEW

**Comment by tgravescs** [comment]
@jiangxb1987 @mengxr

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala [hunk 1]
```diff
@@ -68,13 +68,13 @@ private object ResourceRequestHelper extends Logging {
       (AM_CORES.key, YARN_AM_RESOURCE_TYPES_PREFIX + "cpu-vcores"),
       (DRIVER_CORES.key, YARN_DRIVER_RESOURCE_TYPES_PREFIX + "cpu-vcores"),
       (EXECUTOR_CORES.key, YARN_EXECUTOR_RESOURCE_TYPES_PREFIX + "cpu-vcores"),
-      (s"${SPARK_EXECUTOR_RESOURCE_PREFIX}fpga${SPARK_RESOURCE_COUNT_SUFFIX}",
+      (s"${SPARK_EXECUTOR_RESOURCE_PREFIX}fpga${SPARK_RESOURCE_AMOUNT_SUFFIX}",
         s"${YARN_EXECUTOR_RESOURCE_TYPES_PREFIX}${YARN_FPGA_RESOURCE_CONFIG}"),
-      (s"${SPARK_DRIVER_RESOURCE_PREFIX}fpga${SPARK_RESOURCE_COUNT_SUFFIX}",
+      (s"${SPARK_DRIVER_RESOURCE_PREFIX}fpga${SPARK_RESOURCE_AMOUNT_SUFFIX}",
         s"${YARN_DRIVER_RESOURCE_TYPES_PREFIX}${YARN_FPGA_RESOURCE_CONFIG}"),
       (s"${SPARK_EXECUTOR_RESOURCE_PREFIX}gpu{SPARK_RESOURCE_COUNT_SUFFIX}",
```
Comment (jiangxb1987) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala]: Also update this line?

Comment (tgravescs) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala]: Oh, good catch, that one is actually missing a $.  I used intellij refactor and it built so I was wondering how that happened.

Comment (tgravescs) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala]: Updated and added another test

**Review by jiangxb1987 — APPROVED** [review]
LGTM

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala [hunk 2]
```diff
@@ -390,7 +390,7 @@ class ClientSuite extends SparkFunSuite with Matchers {
     }
   }
 
-  test(s"custom resource request yarn config and spark config fails") {
+  test(s"custom driver resource request yarn config and spark config fails") {
```
Comment (dongjoon-hyun) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala]: nit. Shall we remove `s` from `s"`?

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala [hunk 3]
```diff
@@ -400,19 +400,43 @@ class ClientSuite extends SparkFunSuite with Matchers {
...
 
     assert(error.contains("Do not use spark.yarn.driver.resource.yarn.io/fpga," +
-      " please use spark.driver.resource.fpga"))
+      " please use spark.driver.resource.fpga.amount"))
     assert(error.contains("Do not use spark.yarn.driver.resource.yarn.io/gpu," +
-      " please use spark.driver.resource.gpu"))
+      " please use spark.driver.resource.gpu.amount"))
   }
 
+  test(s"custom executor resource request yarn config and spark config fails") {
```
Comment (dongjoon-hyun) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala]: ditto.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM except two nit comments on test case names.

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala (hunk 3 above)
Comment (tgravescs) [reply resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala]: updated

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala (hunk 2 above)
Comment (tgravescs) [reply resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala]: updated

**Comment by tgravescs** [comment]
Updated, thanks for the reviews, I'll let this run through jenkins and commit if everything looks good.

**Comment by tgravescs** [comment]
very odd test faiure, it has 4 executors instead of 3 which has nothing to do with this change just running local-cluster mode.  I'll kick it again.

**Comment by tgravescs** [comment]
test this please

**Comment by dongjoon-hyun** [comment]
Thank you for updating, @tgravescs . Yes. The failure looks weird. Let's see the new Jenkins result.

**Comment by tgravescs** [comment]
merged to master

**Pull request closed without merging**